andela/Ah-backend-aquaman
# #164069214 Document API using Swagger and coreapi

**What does this PR do?**

This PR enables users to interact with the documented API endpoints through the web browser. The documentation is done using both swagger and coreapi

**Description of Task to be completed?**

**For swagger documentation:**
- Install django-rest-swagger and drf-yasg
- Add a url to the documentation in the main urls file
- Add swagger settings to the settings file
- Add rest_framework_swagger and drf_yasg to installed apps in the settings file
- Modify the views file to accommodate the swagger documentation

**For coreapi:**
- Install coreapi through pip install coreapi
- Import module include_docs_urls from rest_framework.documentation
- Include a url to the documentation in the main urls file

**How should this be manually tested?**

- Clone this repo and cd into Ah-backend-aquaman
- Check out the branch using git checkout ft-document-api-164069214
- Create and activate your virtual environment
- Then install the necessary packages using pip install -r requirements.txt
- Then run python manage.py makemigrations
- Followed by python manage.py migrate
- Then run the server python manage.py runserver
- Then copy the link displayed, paste it in your browser and add apidocs/ to use swagger documentation or docs/ to use the coreapi documentation
- The link you  paste should look like this: http://127.0.0.1:8000/apidocs/ or http://127.0.0.1:8000/docs/ 

**What are the relevant pivotal tracker stories?**

[#164069214](https://www.pivotaltracker.com/story/show/164069214)

**Screenshots**

The screenshot shows the documented endpoints that the user can see and interact with after successfully entering the link in their web browser.

**Using Swagger:**
![image](https://user-images.githubusercontent.com/42435346/53752982-2b71cf80-3ec1-11e9-89a6-89fee3457096.png)

**Using coreapi:**
![image](https://user-images.githubusercontent.com/42435346/53862270-e00e0d00-3ff7-11e9-815b-fc7186b8c311.png)
